lib/shoulda/matchers/action_controller/callback_matcher.rb

```diff
@@ -188,8 +321,38 @@ def callbacks
           end
         end
 
+        def filter_matches?
+          callbacks.map(&:filter).include?(method_name)
+        end
+
+        def only_matches?
+          if options.key?(:only)
+            callbacks.any? do |c|
+              c.filter == method_name && callback_if(c) == only_options_as_if
+            end
+          else
+            true
+          end
+        end
+
+        def callback_if(callback)
+          callback.instance_variable_get(:@if).first
```

Comment:
Can a callback store multiple `if` expressions? Is it best to only grab the first `if`, or should we be considering all of them?